tests/algorithms/test_snpe_posterior.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+import torch
+import random
+
+import sbibm
+from sbibm.algorithms.sbi.snpe import run as run_posterior
+from sbibm.metrics.c2st import c2st
+
+#a fast test
+@pytest.mark.parametrize(
+    "task_name,num_observation",
+    [
+        (task_name, num_observation)
+        for task_name in ["gaussian_linear", "slcp",]
+        for num_observation in random.sample(range(1, 11), 2)
```

Comment:
I'd prefer keeping the tests deterministic, i.e., let's just run them on observation 1 and 2 only if 10 is too slow. How long does this test take on your machine?

Reply:
With 2 observations and 2.500 simulated samples per task, all 6 tests in total run for 220s on my box. That is still quite long and mostly due to NPE running epochs. In this case, this is a hen-and-egg problem: we balance simulation budget versus expressiveness of the predictor.
I'd propose to ditch the SLCP task or at least one gaussian task. Limiting the number of epochs would be most ideal - but impossible I guess at the moment.

Reply:
I boiled it down to 2 tasks and about 140s to complete. Still quite long. I guess the central question is, what we want to achieve with this test. If it is really only about the interface, then one task with one observation is enough.

Reply:
Yes, I think we only want to do interface testing as part of `sbibm`. I have just merged a PR that adds an argument to be able to set `max_num_epochs` explicitly. How about running the test for a low maximum number of epochs and only checking that the shape of the return is correct?

Reply:
Cool, these tests are now going at 30s on my box. That is still long, but at least `c2st` appears to return something meaningful. e642d77